account_payment_term_extension/models/account_move.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Tecnativa Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _recompute_payment_terms_lines(self):
+        context_prev = dict(self.env.context)
+        self.env.context = context_prev
```

Comment:
Why is this assignment necessary?

Reply:
Really, odoo use `_compute_payment_terms` (https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_move.py#L880) to define amount and after use `compute` function (https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_payment_term.py#L34) in payment term.
IMO, use `amount_untaxed` from invoice need to add invoice to context to use it later because other possibilities change code so much and can't reused to other addons.

Reply:
Yes, I undertand that. What I meant was that, in line 13, you are assigning `context_prev` to `self.env.context`, but `context_prev` was already `self.env.context`, right? That seems unnecessary... Or am I missing something?

Reply:
I use that to leave the context as it was later.

Reply:
This is not correct. All contexts changes must be done through `with_context`, but I don't think you need some context handling for doing this, but instead overwriting the proper method.

Reply:
I have changed it to `with_context` now. I have thought about some possible solutions but IMO it's the best way to improve without change everything. Can you propose other better solution?

Reply:
AFAIK, whole `compute` method was overridden in this module, so you just need to add the `if` part in that method.

Reply:
Currently `compute_line_amount` is already used but to that function we must send an amount that we do not have (`amount_untaxed` of the invoice).

The `compute` function in `account.payment.term` model is very useful but it gets amonunt of the total of the invoice to calculate all the lines and this is insufficient since it is necessary to keep that value for the balance calculation but the amount_untaxed of the invoice is also needed to use it in some line. For this reason `with_contect()` is used.